django/utils/dateformat.py

```diff
@@ -30,11 +30,17 @@
 
 
 class Formatter(object):
+
+    def format_piece(self, piece):
```

Comment:
Maybe `validate_piece` would be a better name since no formatting is happening as might be implied by the current name?